src/core/database.py

```diff
@@ -0,0 +1,78 @@
+# coding: utf8
+
+__contact__ = "info@hytech-imaging.fr"
+__copyright__ = "Copyright (c) 2021 Hytech Imaging"
+
+import os.path
+
+from PyQt5.QtWidgets import QMessageBox
+from qgis._core import (
+    QgsWkbTypes,
+    QgsFields,
+    QgsVectorFileWriter,
+    QgsCoordinateReferenceSystem,
+    QgsCoordinateTransformContext,
+    QgsFeatureSink,
+    QgsVectorLayer,
+    QgsProject,
+)
+
+CONST_DB_NAME = "sammo-boat.gpkg"
+CONST_LAYER_NAME = "session datas"
+
+class SammoDataBase:
+    def isDataBaseAvailableInThisDirectory(self, directory):
+        return os.path.isfile(self._getPathToDataBase(directory))
+
+    def getDbName(self):
```

Comment:
If we want to be pythonic, we shouldn't have these kind of getter/setter. Instead, you can use property. In case of a getter:

``` python
@property
def dbName(self):
    return DB_NAME
```

This way we can just do:

``` python
d = SammoDatabase()
d.dbName
```

Reply:
Done commit #8 (33fc7e1847da170fdcc5362c7978609aace3250c)